Src/ILGPU/IR/Types/StructureType.cs

```diff
@@ -939,6 +943,17 @@ internal Type GetDefaultManagedType<TTypeProvider>(TTypeProvider typeProvider)
             TTypeProvider typeProvider) =>
             typeProvider.GetStructureType(this);
 
+        /// <summary cref="TypeNode.Write{T}(T)"/>
+        protected internal override void Write<T>(T writer)
+        {
+            int index = 0;
+            writer.Write(nameof(NumFields), NumFields);
+            foreach (var fieldType in Fields)
+            {
```

Comment:
NIT: Please remove braces that are not required to match the style of the code base